test/extended/util/volumes.go

```diff
@@ -0,0 +1,219 @@
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	kapiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// CreatePersistentVolume creates a HostPath Persistent Volume.
+func CreatePersistentVolume(name, capacity, hostPath string) *kapiv1.PersistentVolume {
+	return &kapiv1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapiv1.PersistentVolumeSpec{
+			PersistentVolumeSource: kapiv1.PersistentVolumeSource{
+				HostPath: &kapiv1.HostPathVolumeSource{
+					Path: hostPath,
+				},
+			},
+			Capacity: kapiv1.ResourceList{
+				kapiv1.ResourceStorage: resource.MustParse(capacity),
+			},
+			AccessModes: []kapiv1.PersistentVolumeAccessMode{
+				kapiv1.ReadWriteOnce,
+				kapiv1.ReadOnlyMany,
+				kapiv1.ReadWriteMany,
+			},
+		},
+	}
+}
+
+// SetupHostPathVolumes will create the requested number of persistent volumes with the given capacity
+func SetupHostPathVolumes(oc *CLI, capacity string, count int) (volumes []*kapiv1.PersistentVolume, err error) {
+	c := oc.AdminKubeClient().Core().PersistentVolumes()
+	prefix := oc.Namespace()
+
+	rootDir, err := ioutil.TempDir(TestContext.OutputDir, "persistent-volumes")
+	if err != nil {
+		e2e.Logf("Error creating pv dir %s: %v\n", TestContext.OutputDir, err)
+		return volumes, err
+	}
+	e2e.Logf("Created pv dir %s\n", rootDir)
+	for i := 0; i < count; i++ {
+		dir, err := ioutil.TempDir(rootDir, fmt.Sprintf("%0.4d", i))
+		if err != nil {
+			e2e.Logf("Error creating pv subdir %s: %v\n", rootDir, err)
+			return volumes, err
+		}
+		e2e.Logf("Created pv subdir %s\n", dir)
+		if _, err = exec.LookPath("chcon"); err == nil {
+			e2e.Logf("Found chcon in path\n")
+			//err := exec.Command("chcon", "-t", "container_file_t", dir).Run()
```

Comment:
delete?

Reply:
That was in the original method that I re-homed, but i'll go ahead and remove it.